src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesView/ExternalDataSourcesView.js

```diff
@@ -173,109 +202,122 @@ const ExternalDataSourcesView = ({
-              />
-            </Col>
-          </Row>
-        </Layout>
         <Row>
           <Col xs={3}>
             <KeyValue
               data-test-external-data-source-listprefix
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.listPrefix" />}
-              value={value?.listPrefix ?? <NoValue />}
+              value={externalDataSource?.listPrefix ?? <NoValue />}
             />
           </Col>
           <Col xs={5}>
             <KeyValue
               data-test-external-data-source-fullprefix
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.fullPrefix" />}
-              value={value?.fullPrefix ?? <NoValue />}
+              value={externalDataSource?.fullPrefix ?? <NoValue />}
             />
           </Col>
           <Col xs={4}>
             <KeyValue
               data-test-external-data-source-principal
               label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.principal" />}
-              value={value?.principal ?? <NoValue />}
+              value={externalDataSource?.principal ?? <NoValue />}
             />
           </Col>
         </Row>
         <KeyValue
           data-test-external-data-source-credentials
           label={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.credentials" />}
-          value={value?.credentials ?? <NoValue />}
+          value={externalDataSource?.credentials ?? <NoValue />}
+        />
+      </Pane>
+      {deleteModal && (
+        <ConfirmationModal
+          buttonStyle="danger"
+          confirmLabel={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.delete.confirmLabel" />}
+          data-test-confirmationModal
+          heading={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.delete.confirmHeading" />}
+          id="delete-external-data-source-confirmation"
+          message={<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.delete.confirmMessage" values={{ name: externalDataSource?.name }} />}
+          onCancel={() => setDeleteModal(false)}
+          onConfirm={() => {
+            onDelete(externalDataSource?.id);
+            onClose();
+            setDeleteModal(false);
+          }}
+          open={deleteModal}
         />
-      </Card>
+      )}
+      <ExternalDataSourcesFormModal
+        initialValues={mode === EDITING ?
+          { ...externalDataSource } :
+          {
+            active: false,
+            activationEnabled: false,
+            rectype: 1,
+            supportsHarvesting: true,
+            type: 'org.olf.kb.adapters.GOKbOAIAdapter',
+          }}
+        modalProps={{
+          dismissible: true,
+          label: <FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.edit" values={{ name: externalDataSource?.name }} />,
+          onClose: () => setMode(VIEWING),
```

Comment:
Same thing as above, we can have a single state in _this_ component managing true/false as to whether this modal is open, and then we won't have to pass state down from Settings component at all